pipeline.cpp

```diff
@@ -80,7 +81,42 @@ void DrawLine(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* cons
  ************************************************************/
 void DrawTriangle(Buffer2D<PIXEL> & target, Vertex* const triangle, Attributes* const attrs, Attributes* const uniforms, FragmentShader* const frag)
 {
-    // Your code goes here
+    // Find a bounding box for the triangle
+    int minx = std::min({triangle[0].x, triangle[1].x, triangle[2].x});
+    int miny = std::min({triangle[0].y, triangle[1].y, triangle[2].y});
+    int maxx = std::max({triangle[0].x, triangle[1].x, triangle[2].x});
+    int maxy = std::max({triangle[0].y, triangle[1].y, triangle[2].y});
+    
+    // Loop through every pixel in the bounding box
+    for(int y = miny; y < maxy; y++)
+        for(int x = minx; x < maxx; x++)
+        {
+            // Side 1
```

Comment:
I'm guessing this bit here is how you got your sides smooth on your triangles. I couldn't quite figure that out. good job.